### ISSUE TITLE
The functions you are calling could change at any time

### ISSUE THREAD

**Issue by w0rp** [andreypopp/asyncomplete-ale.vim#1]
I had a look through the functions you're calling, and there are quite a few functions in the ALE codebase that are not part of the public API. You can call them, but changes in ALE could break your code at any time.

Eventually ALE will offer a public API for completion sources. See here: https://github.com/w0rp/ale/pull/2143

**Comment by andreypopp** [comment]
@w0rp thanks for reviewing this!

Yeah, definitely this is a dirty hack and I understand that it can break anytime but this haas unblocked me in the meantime for my migration to LSP-based linters/completions/....

w0rp/ale#2143 looks great and I wanted to propose to expose other LSP method in ALE. I think ALE is a great cornerstone experience for LSP — it syncs docs and lints but UI for other features are too opinionated sometimes and thus I'd prefer to be able to add my own plugins (like asyncomplete here).

**Comment by w0rp** [comment]
Yeah, that's the intention. With a few changes, people can integrate the LSP message part with other completion plugins.
